TROUBLESHOOTING.md

```diff
@@ -0,0 +1,420 @@
+gcloud services enable iam.googleapis.com --project "$SEED_PROJECT"
+```
+
+#### Missing API: `appengine.googleapis.com`
+
+**Error message:**
+
+```
+$ gcloud app describe --project [target-project-id] --format=json
+
+API [appengine.googleapis.com] not enabled on project [[seed-project-number]].
+Would you like to enable and retry (this will take a few minutes)?
+(y/N)?
+ERROR: (gcloud.app.describe) User [[service-account]] does not have permission to access app [[target-project-id]] (or it may not exist): App Engine Admin API has not been used in project [seed-project-number] before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/appengine.googleapis.com/overview?project=[seed-project-number] then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
+- '@type': type.googleapis.com/google.rpc.Help
+  links:
+  - description: Google developers console API activation
+    url: https://console.developers.google.com/apis/api/appengine.googleapis.com/overview?project=[seed-project-number]
+```
+
+**Cause**:
+
+The App Engine API is not enabled on the seed project, which prevents creation of an App Engine instance on the target project.
+
+**Solution:**
+
+```
+# Requires `roles/serviceusage.serviceUsageAdmin` on $SEED_PROJECT
+gcloud services enable appengine.googleapis.com --project "$SEED_PROJECT"
+```
+
+**Notes:**
+
+The absence of the `appengine.googleapis.com` API will not cause Terraform to fail, but any interactions with the App Engine instance will fail. Once this API is activated, it might take a few minutes for Terraform to detect that the App Engine instance is not present and then create it.
+
+- - -
+### Service account missing roles
+
+* Organizational roles
+    * [roles/resourcemanager.organizationViewer](#missing-org-role-roles-resourcemanager-organizationviewer)
+    * [roles/resourcemanager.projectCreator](#missing-org-role-roles-resourcemanager-projectcreator)
+    * [roles/compute.xpnAdmin](#missing-shared-vpc-roles) (when using a shared VPC)
+    * [roles/compute.networkAdmin](#missing-shared-vpc-roles) (when using a shared VPC)
+* Shared VPC project roles (when using a shared VPC)
+    * [roles/resourcemanager.projectIamAdmin](#missing-shared-vpc-roles)
+    * [roles/browser](#missing-shared-vpc-roles)
+* Billing account roles
+*   * [roles/billing.user](#missing-roles-billing-user-role)
+
+#### Missing org role: `roles/resourcemanager.organizationViewer`
+
+**Error message:**
+
+```
+* module.project-factory.data.google_organization.org: 1 error(s) occurred:
+
+* module.project-factory.data.google_organization.org: data.google_organization.org: Error reading Organization Not Found : [organization-id]: googleapi: Error 403: The caller does not have permission, forbidden
+```
+
+**Cause:**
+
+The active service account does not have `roles/resourcemanager.organizationViewer` on the active organization.
```

Comment:
We should also include an explanation of why each role is required, and link to this explanation from the README section.